### PR TITLE
Feature/map legend

### PR DESCRIPTION
**Pull request by thill02**

### CODE REVIEW

File: src/app-pages/explorer/map-legend.js [hunk 1]
```diff
@@ -1,40 +1,35 @@
...
-                    cy='8'
-                    r='5'
-                  />
-                </svg>
-                {x.title}
-              </p>
-            ))}
-          </div>
+
+const status = [
```
Comment (KevinJJackson) [inline src/app-pages/explorer/map-legend.js:3]: we probably don't want to have to update this if new statuses do come up, looks like you could use some workarounds in react to manage when things like this change. something like this should fix the reloading bug you were having, but might need to come up with another sorting solution still:

```js
    const [statuses, setStatuses] = useState([]);

    useEffect(() => {
      const { status } = domains;

      if (domains && status) {
        const newStatuses = status.map(s => ({
          title: s.value[0].toUpperCase() + s.value.substr(1).toLowerCase(),
          value: s.value,
        }));

        setStatuses(newStatuses);
      }
    }, [domains, setStatuses]);
```

then just map over `statuses` in the card-body

Comment (KevinJJackson) [reply src/app-pages/explorer/map-legend.js:3]: Could do something similar with how you used the ordering with something like below:

```js
    useEffect(() => {
      const { status } = domains;
      const statusOrder = ['Active', 'Lost', 'Inactive', 'Abandoned', 'Destroyed'];

      if (domains && status) {
        const newStatuses = status.map(s => {
          const title = s.value[0].toUpperCase() + s.value.substr(1).toLowerCase();

          return {
            title,
            value: s.value,
            order: statusOrder.indexOf(title),
          };
        });

        newStatuses.sort((a, b) => (a.order > b.order ? 1 : -1));

        setStatuses(newStatuses);
      }
    }, [domains, setStatuses]);
```

File: src/app-pages/explorer/map-legend.js [hunk 2]
```diff
@@ -1,40 +1,35 @@
...
+  { title: 'Lost', value: 'lost', order: 2 }
+];
+const MapLegend = () => {
+  status.sort((a, b) => a.order > b.order ? 1 : -1);
+  return (
+    <div style={{ position: 'absolute', top: 10, right: 10 }}>
+      <div className='card' style={{ opacity: 0.8 }}>
+        <div className='card-body'>
+          {status.map((x, i) => (
+            <p className='mb-2 mt-2' key={i}>
```
Comment (KevinJJackson) [inline src/app-pages/explorer/map-legend.js:17]: using the index of a map function is usually considered bad practice as a key, you should use something that is constant and defines the element. in this case `status.title` works very well as a key

**Comment by thill02** [comment]
https://github.com/USACE/instrumentation/issues/60

**Comment by thill02** [comment]
https://github.com/USACE/instrumentation/issues/60

**Pull request merged**